### PR TITLE
updated ClipboardItem and Clipboard documentation and examples using …

### PR DESCRIPTION
**Pull request by tayloregivens**
…supports()

Updated the web API ClipboardItem and Clipboard pages to describe the supports() method. I also updated the code samples that write data to the clipboard so they use the supports method as well.

My team at Microsoft created the supports() static method and so we wanted to go ahead and document on MDN as well.

### CODE REVIEW

File: files/en-us/web/api/clipboard/write/index.md [hunk 1]
```diff
@@ -54,14 +54,18 @@ Note that for this particular case, you could just as readily use `Clipboard.wri
...
+    const data = [new ClipboardItem({ [type]: blob })];
+    await navigator.clipboard.write(data);
+  } else {
+    console.log("text/plain is not supported");
+  }
 }
 ```
 
-The `setClipboard()` method begins by creating a new a {{domxref("Blob")}} object.
+The `setClipboard()` method begins with an if statement that returns true or false based on whether the clipboard supports 'text/plain' MIME type. If the clipboard does support the MIME type, then a new {{domxref("Blob")}} object is created.
```
Comment (snianu) [inline files/en-us/web/api/clipboard/write/index.md:68]: Nit: Can we make it a little more succinct? e.g.
`The `setClipboard()` method first checks for the support of `text/plain` MIME type. If it's supported, a new {{domxref("Blob")}} object is created.`
Please feel free to rephrase this to make it more concise.

File: files/en-us/web/api/clipboard/write/index.md [hunk 2]
```diff
@@ -81,12 +85,16 @@ const canvas = document.getElementById("canvas");
 canvas.addEventListener("click", copyCanvasContentsToClipboard);
 
 async function copyCanvasContentsToClipboard() {
```
Comment (snianu) [inline files/en-us/web/api/clipboard/write/index.md:87]: Can we change the below sentence in ln 80 to describe the changes we made in this example?
> This example writes the canvas to a blob, using the default MIME type of `image/png`, and then writes the blob to the clipboard.

Here we can mention that the author doesn't have to generate the data for images if the `image/png` type isn't supported by the browser.

File: files/en-us/web/api/clipboarditem/index.md [hunk 3]
```diff
@@ -27,6 +27,13 @@ _This interface provides the following properties._
 - {{domxref("ClipboardItem.presentationStyle", "presentationStyle")}} {{ReadOnlyInline}}
   - : Returns one of the following: `"unspecified"`, `"inline"` or `"attachment"`.
 
+## Static methods
+
+_This interface defines the following methods._
+
+- {{domxref("ClipboardItem.supports", "supports()")}}
+  - : Returns true or false whether a {{Glossary("MIME type")}} type is supported. This allows us to detect if a {{Glossary("MIME type")}} is supported by the clipboard before attempting to write data.
```
Comment (snianu) [inline files/en-us/web/api/clipboarditem/index.md:35]: Nit: _before attempting to generate the data for the {{Glossary("MIME type")}} and write to the clipboard_.

File: files/en-us/web/api/clipboarditem/index.md [hunk 4]
```diff
@@ -38,22 +45,25 @@ _This interface defines the following methods._
 
 ### Writing to the clipboard
 
-Here we're writing a new {{domxref("ClipboardItem.ClipboardItem", "ClipboardItem()")}} to the system clipboard by requesting a PNG image using the {{domxref("Fetch API")}}, and in turn, the {{domxref("Response.blob()", "responses' blob()")}} method, to create the new `ClipboardItem`.
+Here we're writing a new {{domxref("ClipboardItem.ClipboardItem", "ClipboardItem()")}} to the system clipboard by requesting a PNG image using the {{domxref("Fetch API")}}, and in turn, the {{domxref("Response.blob()", "responses' blob()")}} method, to create the new `ClipboardItem`. Then we use {{domxref("ClipboardItem.supports", "supports()")}} to determine if the MIME data type is supported before writing it to the clipboard.
```
Comment (snianu) [inline files/en-us/web/api/clipboarditem/index.md:48]: Looks like we changed it to first check for the support of the PNG image MIME type and _then_ generate the data and write to the clipboard, right? Can we change this text to reflect the changes we made here?

File: files/en-us/web/api/clipboarditem/supports/index.md [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    }
+  } catch (err) {
+    console.error(err.name, err.message);
+  }
+}
+```
+
+## Specifications
+
+{{Specifications}}
```
Comment (snianu) [inline files/en-us/web/api/clipboarditem/supports/index.md:56]: Spec link for `supports`: https://w3c.github.io/clipboard-apis/#dom-clipboarditem-supports

**Review by snianu — COMMENTED** [review]
Added some comments. PTAL, thanks!

File: files/en-us/web/api/clipboarditem/supports/index.md (hunk 5 above)
Comment (captainbrosset) [reply files/en-us/web/api/clipboarditem/supports/index.md:56]: This link should come from BCD. The `{{Specifications}}` macro will pull it in automatically if it's present in BCD (browser-compat-data).

File: files/en-us/web/api/clipboarditem/supports/index.md [hunk 6]
```diff
@@ -0,0 +1,65 @@
+---
+title: "ClipboardItem: supports() method"
+short-title: supports()
+slug: Web/API/ClipboardItem/supports
+page-type: web-api-static-method
+browser-compat: api.ClipboardItem.supports
```
Comment (captainbrosset) [inline files/en-us/web/api/clipboarditem/supports/index.md:6]: I think this should be:

```suggestion
browser-compat: api.ClipboardItem.supports_static
```

to match the BCD data: https://github.com/mdn/browser-compat-data/blob/main/api/ClipboardItem.json#L173

File: files/en-us/web/api/clipboarditem/supports/index.md (hunk 5 above)
Comment (captainbrosset) [reply files/en-us/web/api/clipboarditem/supports/index.md:56]: Just verified, the data is present: https://github.com/mdn/browser-compat-data/blob/main/api/ClipboardItem.json#L175C24-L175C88

File: files/en-us/web/api/clipboard/writetext/index.md [hunk 7]
```diff
@@ -45,7 +45,11 @@ button.addEventListener("click", () => writeClipboardText("<empty clipboard>"));
 
 async function writeClipboardText(text) {
   try {
-    await navigator.clipboard.writeText(text);
+    if (ClipboardItem.supports("text/plain")) {
```
Comment (captainbrosset) [inline files/en-us/web/api/clipboard/writetext/index.md:48]: I wonder if we really need to change examples that deal with plain text. Are there cases where plain text is **not** supported?

File: files/en-us/web/api/clipboarditem/supports/index.md [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+- `type`
+  - : A valid {{Glossary("MIME type")}}.
+
+### Return value
+
+True or false value based on whether that {{Glossary("MIME type")}} is supported by the clipboard.
+
+## Examples
+
+In the following example, we're writing data to the clipboard via the {{domxref("clipboard.write()")}} method. Then we utilize {{domxref("ClipboardItem.supports", "supports()")}} to see if the {{Glossary("MIME type")}} is supported by the clipboard before attempting to write it to the clipboard. Using {{domxref("ClipboardItem.supports", "supports()")}} saves us from costly and unesseary CPU cycles from attempting to write unsupported formats to the clipboard. A `try..catch` block can be used to catch any errors while writing the data to the clipboard.
```
Comment (wbamberg) [inline files/en-us/web/api/clipboarditem/supports/index.md:30]:
```suggestion
### Writing an image to the clipboard

In the following example, we're writing an image to to the clipboard.

We use `supports()` to check whether the `"image/png"` MIME type is supported by the clipboard before fetching the image and writing it using {{domxref("clipboard.write()")}}.

We also use a [`try..catch`](/en-US/docs/Web/JavaScript/Reference/Statements/try...catch) statement to catch any errors while writing the data to the clipboard.
```
 
- use a descriptive title for the example, and a sentence giving an overview of what we are doing
- don't self-link with `{{domxref("ClipboardItem.supports", "supports()")}}`
- avoid passive voice in the bit about `try...catch`, and link to the JS docs

I wasn't sure about "Using {{domxref("ClipboardItem.supports", "supports()")}} saves us from costly and unnecessary CPU cycles from attempting to write unsupported formats to the clipboard." I think as a rule examples should just illustrate, that is they "show not tell" so things about the motivation for the method should live in the method general description. But I'm not sure this is needed there either, perhaps it is too obvious?

File: files/en-us/web/api/clipboarditem/supports/index.md [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+
+```js
+async function writeClipImg() {
```
Comment (wbamberg) [inline files/en-us/web/api/clipboarditem/supports/index.md:33]:
```suggestion
async function writeImageToClipboard() {
```

...this is strictly optional but I like more readable method names.

File: files/en-us/web/api/clipboard/writetext/index.md (hunk 7 above)
Comment (wbamberg) [reply files/en-us/web/api/clipboard/writetext/index.md:48]: If I read the spec right, no: https://w3c.github.io/clipboard-apis/#mandatory-data-types.

File: files/en-us/web/api/clipboarditem/index.md (hunk 3 above)
Comment (wbamberg) [reply files/en-us/web/api/clipboarditem/index.md:35]:
```suggestion
  - : Checks whether a given {{Glossary("MIME type")}} is supported by the clipboard. This allows us to detect if a MIME type is supported by the clipboard before attempting to write data.
```

(although I would personally omit the second sentence and have it in the main page)

File: files/en-us/web/api/clipboarditem/supports/index.md (hunk 9 above)
Comment (wbamberg) [reply files/en-us/web/api/clipboarditem/supports/index.md:33]: Oh I see the other examples use `writeClipImg` as well. So I guess this is even more optional :).

File: files/en-us/web/api/clipboard/write/index.md (hunk 1 above)
Comment (wbamberg) [reply files/en-us/web/api/clipboard/write/index.md:68]: I would say something like:

```suggestion
The `setClipboard()` method first uses the {{domxref("ClipboardItem.supports()")}} static method to check whether the clipboard supports the `"text/plain"` MIME type.

If the clipboard does support plain text, then `setClipboard()` creates a new {{domxref("Blob")}} object.
```

But as Patrick said, is it good practice to use `supports()` for plain text?

File: files/en-us/web/api/clipboard/write/index.md [hunk 10]
```diff
@@ -54,14 +54,18 @@ Note that for this particular case, you could just as readily use `Clipboard.wri
 button.addEventListener("click", () => setClipboard("<empty clipboard>"));
 
 async function setClipboard(text) {
-  const type = "text/plain";
-  const blob = new Blob([text], { type });
-  const data = [new ClipboardItem({ [type]: blob })];
-  await navigator.clipboard.write(data);
+  if (ClipboardItem.supports("text/plain")) {
+    const type = "text/plain";
```
Comment (wbamberg) [inline files/en-us/web/api/clipboard/write/index.md:58]: It's a bit weird to use a literal for `text/plain`, then declare a variable for it in the next line. Perhaps:

```suggestion
  const type = "text/plain";
  if (ClipboardItem.supports(type)) {
```

?

**Review by wbamberg — DISMISSED** [review]
Thank you for the PR, @tayloregivens !

**Comment by wbamberg** [comment]
Oh, also I missed that the directory for `supports` should have the `_static` suffix. See for example `json_static` or `error_static` in https://github.com/mdn/content/tree/main/files/en-us/web/api/response. We need to do this to support static and instance methods that have the same name, like https://developer.mozilla.org/en-US/docs/Web/API/Response/json_static and https://developer.mozilla.org/en-US/docs/Web/API/Response/json.

File: files/en-us/web/api/clipboarditem/supports_static/index.md [hunk 11]
```diff
@@ -0,0 +1,71 @@
+---
+title: "ClipboardItem: supports() static method"
+short-title: supports()
+slug: Web/API/ClipboardItem/supports
```
Comment (wbamberg) [inline files/en-us/web/api/clipboarditem/supports_static/index.md:4]:
```suggestion
slug: Web/API/ClipboardItem/supports_static
```

...hopefully this will fix the CI error.

File: files/en-us/web/api/clipboard/write/index.md [hunk 12]
```diff
@@ -55,13 +55,19 @@ button.addEventListener("click", () => setClipboard("<empty clipboard>"));
...
+    const data = [new ClipboardItem({ [type]: blob })];
+    await navigator.clipboard.write(data);
+  } else {
+    console.log("text/plain is not supported");
+  }
 }
 ```
 
-The `setClipboard()` method begins by creating a new a {{domxref("Blob")}} object.
+The `setClipboard()` method first uses the {{domxref("ClipboardItem.supports()")}} static method to check whether the clipboard supports the `"text/plain"` MIME type.
```
Comment (wbamberg) [inline files/en-us/web/api/clipboard/write/index.md:68]:
```suggestion
The `setClipboard()` method first uses the {{domxref("ClipboardItem.supports_static", "ClipboardItem.supports()")}} static method to check whether the clipboard supports the `"text/plain"` MIME type.
```

(domxref is not very nice to use with statics. Hopefully this ought to fix the flaws reported for this page)

File: files/en-us/web/api/clipboarditem/index.md [hunk 13]
```diff
@@ -27,6 +27,13 @@ _This interface provides the following properties._
 - {{domxref("ClipboardItem.presentationStyle", "presentationStyle")}} {{ReadOnlyInline}}
   - : Returns one of the following: `"unspecified"`, `"inline"` or `"attachment"`.
 
+## Static methods
+
+_This interface defines the following methods._
+
+- {{domxref("ClipboardItem.supports", "supports()")}}
```
Comment (wbamberg) [inline files/en-us/web/api/clipboarditem/index.md:34]:
```suggestion
- {{domxref("ClipboardItem.supports_static", "ClipboardItem.supports()")}} 
```

File: files/en-us/web/api/clipboarditem/index.md [hunk 14]
```diff
@@ -38,21 +45,24 @@ _This interface defines the following methods._
 
 ### Writing to the clipboard
 
-Here we're writing a new {{domxref("ClipboardItem.ClipboardItem", "ClipboardItem()")}} to the system clipboard by requesting a PNG image using the {{domxref("Fetch API")}}, and in turn, the {{domxref("Response.blob()", "responses' blob()")}} method, to create the new `ClipboardItem`.
+Here we use {{domxref("ClipboardItem.supports", "supports()")}} to determine if 'img/png' MIME data type is supported before generating the image data and writing it to the clipboard. The image is generated using the {{domxref("Fetch API")}}, and in turn, the {{domxref("Response.blob()", "responses' blob()")}} method is used to create a `ClipboardItem` that is written to the clipboard.
```
Comment (wbamberg) [inline files/en-us/web/api/clipboarditem/index.md:48]:
```suggestion
Here we use {{domxref("ClipboardItem.supports", "supports()")}} to determine if 'img/png' MIME data type is supported before generating the image data and writing it to the clipboard. The image is generated using the {{domxref("Fetch API")}}, and in turn, the {{domxref("Response.blob()", "responses' blob()")}} method is used to create a `ClipboardItem` that is written to the clipboard.
```
```suggestion
Here we use {{domxref("ClipboardItem.supports_static", "supports()")}}  to determine if 'img/png' MIME data type is supported before generating the image data and writing it to the clipboard. The image is generated using the {{domxref("Fetch API")}}, and in turn, the {{domxref("Response.blob()", "responses' blob()")}} method is used to create a `ClipboardItem` that is written to the clipboard.
```

File: files/en-us/web/api/clipboard/write/index.md [hunk 15]
```diff
@@ -55,13 +55,19 @@ button.addEventListener("click", () => setClipboard("<empty clipboard>"));
 
 async function setClipboard(text) {
   const type = "text/plain";
-  const blob = new Blob([text], { type });
-  const data = [new ClipboardItem({ [type]: blob })];
-  await navigator.clipboard.write(data);
+  if (ClipboardItem.supports(type)) {
```
Comment (captainbrosset) [inline files/en-us/web/api/clipboard/write/index.md:58]: As discussed offline, let's remove this change since `text/plain` is always supported.

File: files/en-us/web/api/clipboard/write/index.md [hunk 16]
```diff
@@ -55,13 +55,19 @@ button.addEventListener("click", () => setClipboard("<empty clipboard>"));
...
+    const data = [new ClipboardItem({ [type]: blob })];
+    await navigator.clipboard.write(data);
+  } else {
+    console.log("text/plain is not supported");
+  }
 }
 ```
 
-The `setClipboard()` method begins by creating a new a {{domxref("Blob")}} object.
+The `setClipboard()` method first uses the {{domxref("ClipboardItem.supports_static", "ClipboardItem.supports()")}} static method to check whether the clipboard supports the `"text/plain"` MIME type.
```
Comment (captainbrosset) [inline files/en-us/web/api/clipboard/write/index.md:68]: And revert this change too, to match the code block above.

File: files/en-us/web/api/clipboarditem/index.md [hunk 17]
```diff
@@ -27,6 +27,13 @@ _This interface provides the following properties._
 - {{domxref("ClipboardItem.presentationStyle", "presentationStyle")}} {{ReadOnlyInline}}
   - : Returns one of the following: `"unspecified"`, `"inline"` or `"attachment"`.
 
+## Static methods
+
+_This interface defines the following methods._
+
+- {{domxref("ClipboardItem.supports_static", "ClipboardItem.supports()")}}
+  - : Checks whether a given {{Glossary("MIME type")}} is supported by the clipboard. This allows us to detect if a MIME type is supported by the clipboard before attempting to write data.
```
Comment (captainbrosset) [inline files/en-us/web/api/clipboarditem/index.md:35]:
```suggestion
  - : Checks whether a given {{Glossary("MIME type")}} is supported by the clipboard. This allows to detect if a MIME type is supported by the clipboard before attempting to write data.
```

The usage of "us" felt a bit inconsistent with other reference docs.

Comment (captainbrosset) [reply files/en-us/web/api/clipboarditem/index.md:35]: I see this page uses "us"/"we" in the examples section. I think it's fine there because it's more of a tutorial. But in the reference part, it feels a bit out of place.

**Review by captainbrosset — APPROVED** [review]
Drive-by approval with a couple of comments. Let's wait for Will to do a more official review.

File: files/en-us/web/api/clipboarditem/index.md (hunk 17 above)
Comment (wbamberg) [reply files/en-us/web/api/clipboarditem/index.md:35]: I agree about "us" but "allows to detect" is not quite right. "enables to website to detect" is better if a bit longwinded.

File: files/en-us/web/api/clipboarditem/index.md [hunk 18]
```diff
@@ -38,21 +45,24 @@ _This interface defines the following methods._
 
 ### Writing to the clipboard
 
-Here we're writing a new {{domxref("ClipboardItem.ClipboardItem", "ClipboardItem()")}} to the system clipboard by requesting a PNG image using the {{domxref("Fetch API")}}, and in turn, the {{domxref("Response.blob()", "responses' blob()")}} method, to create the new `ClipboardItem`.
+Here we use {{domxref("ClipboardItem.supports_static", "supports()")}} to determine if 'img/png' MIME data type is supported before generating the image data and writing it to the clipboard. The image is generated using the {{domxref("Fetch API")}}, and in turn, the {{domxref("Response.blob()", "responses' blob()")}} method is used to create a `ClipboardItem` that is written to the clipboard.
```
Comment (wbamberg) [inline files/en-us/web/api/clipboarditem/index.md:48]:
```suggestion
Here we use {{domxref("ClipboardItem.supports_static", "supports()")}} to check whether the 'img/png' MIME data type is supported. If it is, we fetch the image with the ["Fetch API"](/en-US/docs/Web/API/Fetch_API), and then read it into a {{domxref("Blob")}}, which we can use to create a `ClipboardItem` that is written to the clipboard.
```

(however, if we decide not to use `supports()` with "img/png", per https://github.com/mdn/content/pull/33262#discussion_r1585203697, then obviously this won't apply, and we should just revert this part of the PR.)

**Review by wbamberg — COMMENTED** [review]
Thank you for the updates @tayloregivens !

I had just one more suggested change, but I think we also have an open question about whether it's appropriate to use `supports()` with "img/png" (https://github.com/mdn/content/pull/33262#discussion_r1585203697). If not we should use a different MIME type for the `supports()` page and remove the `supports()` check from the examples that use "img/png" as well as the plain text ones.

File: files/en-us/web/api/clipboarditem/supports_static/index.md [hunk 19]
```diff
@@ -0,0 +1,71 @@
...
+## Syntax
+
+```js-nolint
+supports(type)
+```
+
+### Parameters
+
+- `type`
+  - : A string representing a {{Glossary("MIME type")}}.
```
Comment (hamishwillee) [inline files/en-us/web/api/clipboarditem/supports_static/index.md:22]: The spec is not quite this simple. 
- It defines [mandatory types](https://w3c.github.io/clipboard-apis/#mandatory-data-types) and [optional types](https://w3c.github.io/clipboard-apis/#optional-data-types-x).
- For the optional types it defines custom types which are prefixed with `web ` and have a second part that passes a mime-type-format check.

Given the set is small and it isn't just any MIME type, we should list them here

File: files/en-us/web/api/clipboard/write/index.md (hunk 15 above)
Comment (hamishwillee) [reply files/en-us/web/api/clipboard/write/index.md]: 1. We should list the types in supports() as per https://github.com/mdn/content/pull/33262#discussion_r1597862947 - the lists are small and discrete, while the current implication of that doc is you can use any MIME type.
2. I don't understand it yet, but the interesting optional type I'd like here is the custom type prefixed by `web `.

FYI, I'm showing an interest because this is being supported in FF127 and making sure there are docs is obviously part of that https://github.com/mdn/content/issues/33564 . @tayloregivens Great so see work already happening on this!

File: files/en-us/web/api/clipboard/write/index.md [hunk 20]
```diff
@@ -73,20 +73,24 @@ A `try..catch` block could be used to catch any errors writing the data.
 
 ### Write canvas contents to the clipboard
 
-This example writes the canvas to a blob, using the default MIME type of `image/png`, and then writes the blob to the clipboard.
+This example writes the canvas to a blob, using the default MIME type of `image/png`, and then writes the blob to the clipboard. Generating image data is avoided if 'image/png' isn't supported by the browser.
```
Comment (hamishwillee) [inline files/en-us/web/api/clipboard/write/index.md:76]: This feels a bit more natural?

```suggestion
This example first checks if the browser allows 'image/png' data to be written to the clipboard.
If so, it writes the canvas to a blob, using the default MIME type of `image/png`, and then writes the blob to the clipboard. 
```

Given this is a mandatory type the method has to return true right? Seems a bit weird to even bother testing it since if the method is defined it will return true. I'd be tempted to use an optional type for this.

File: files/en-us/web/api/clipboarditem/index.md (hunk 18 above)
Comment (hamishwillee) [reply files/en-us/web/api/clipboarditem/index.md:48]: Ah, this the same problem I see with https://github.com/mdn/content/pull/33262/files#r1597871059 ?  I guess it is OK to for one example to use a mandatory type, but we should have at least one example that does not and ideally cross link.

**Comment by hamishwillee** [comment]
@tayloregivens There are also some merge conflicts here. Let me know if there is anything I can do to help you progress this.

File: files/en-us/web/api/clipboard/write/index.md (hunk 20 above)
Comment (hamishwillee) [reply files/en-us/web/api/clipboard/write/index.md:76]: PS Given this doesn't rely on network, could we perhaps make this a live example?

**Comment by hamishwillee** [comment]
Hi @tayloregivens 
I've fixed up the example/conflicts and rebased (this is now a live example). I'll continue with necessary updates tomorrow, unless you have time to look at them.

File: files/en-us/web/api/clipboarditem/supports_static/index.md [hunk 21]
```diff
@@ -0,0 +1,83 @@
...
+
+### Return value
+
+`true` if the given {{Glossary("MIME type")}} is supported by the clipboard, `false` otherwise.
+
+## Examples
+
+### Writing an image to the clipboard
+
+The following example fetches an SVG image to a blob, and then writes it to the clipboard.
```
Comment (hamishwillee) [inline files/en-us/web/api/clipboarditem/supports_static/index.md:45]: FYI, trivially changed supports to `"image/svg+xml"`. Using `supports()` with image/png should always return true, and is pointless if we know we're always working with PNG. So unless we update this so the example doesn't know what file and hence MIME type it is writing to clipboard in advance, this is better.

File: files/en-us/web/api/clipboarditem/index.md (hunk 18 above)
Comment (hamishwillee) [reply files/en-us/web/api/clipboarditem/index.md:48]: WillB suggestion updated with `image/` rather than `img`, and changed to new type.
I've decided that in this particular case we'll add a note after to indicate that this check isn't strictly necessary.

```suggestion
Here we use {{domxref("ClipboardItem.supports_static", "supports()")}} to check whether the 'image/png' MIME data type is supported. If it is, we fetch the image with the ["Fetch API"](/en-US/docs/Web/API/Fetch_API), and then read it into a {{domxref("Blob")}}, which we can use to create a `ClipboardItem` that is written to the clipboard.
```

**Comment by hamishwillee** [comment]
@tayloregivens I have fixed this up to address the comments. Let's see if we can get it through review!

@wbamberg Can you please look at this again. The main changes are:
1. The `ClipboardItem` and `supports()` page examples use `image/svg+xml` for the mime types.
2. `Clipboard.write()` still uses `image/png` and is a live example (merged in from main and slightly modified). I've added a note to make it clear that the `supports()` check is not really necessary, but highlight some cases where it could be (i.e. showing "for information").
3. The `supports()` method now states the supported MIME types. Note that it omits one in the specification that no one has implemented.

**Comment by tayloregivens** [comment]
> @tayloregivens I have fixed this up to address the comments. Let's see if we can get it through review!
> 
> @wbamberg Can you please look at this again. The main changes are:
> 
> 1. The `ClipboardItem` and `supports()` page examples use `image/svg+xml` for the mime types.
> 2. `Clipboard.write()` still uses `image/png` and is a live example (merged in from main and slightly modified). I've added a note to make it clear that the `supports()` check is not really necessary, but highlight some cases where it could be (i.e. showing "for information").
> 3. The `supports()` method now states the supported MIME types. Note that it omits one in the specification that no one has implemented.

Thanks for addressing the remaining comments and updating the 'supports()' page to use 'image/svg+xml' and making a note that supports() isn't necessary with 'image/png' on the Clipboard.write()

for 3. the supports method allows web authors to specify any arbitrary type and returns false for types that aren't supported by the browser. If you did want to add extra information there you could say that the mandatory types, html, text, and png are always supported and don't require using supports() static method.

File: files/en-us/web/api/clipboarditem/supports_static/index.md [hunk 22]
```diff
@@ -0,0 +1,83 @@
...
+short-title: supports()
+slug: Web/API/ClipboardItem/supports_static
+page-type: web-api-static-method
+browser-compat: api.ClipboardItem.supports_static
+---
+
+{{APIRef("Clipboard API")}} {{securecontext_header}}
+
+The **`supports()`** static method of the {{domxref("ClipboardItem")}} interface indicates whether or not the given {{Glossary("MIME type")}} is supported by the clipboard.
+
```
Comment (hamishwillee) [inline files/en-us/web/api/clipboarditem/supports_static/index.md:12]:
```suggestion
The **`supports()`** static method of the {{domxref("ClipboardItem")}} interface returns `true` if the given {{Glossary("MIME type")}} is supported by the clipboard, and `false` otherwise.

Note that the [Clipboard API](/en-US/docs/Web/API/Clipboard_API) mandates support for plain text, HTML and PNG files.
The `supports()`  method will always return `true` for these MIME types, so testing them is unnecessary .

```

**Review by wbamberg — APPROVED** [review]
👍 thank you @tayloregivens and @hamishwillee for your work on this PR!
